dali/operators/image/resize/experimental/resize_op_impl_cvcuda.h

```diff
@@ -49,37 +52,46 @@ class ResizeOpImplCvCuda : public ResizeBase<GPUBackend>::Impl {
     // effective frames (from videos, channel planes, etc).
     GetResizedShape(out_shape_, in_shape_, make_cspan(params_), 0);
 
-    // Create a map of non-empty samples
-    SetFrameIdxs();
-
     // Now that we know how many logical frames there are, calculate batch subdivision.
     CalculateMinibatchPartition(minibatch_size_);
 
+    CalculateSourceSamples(in_shape, first_spatial_dim);
+
     SetupKernel();
   }
 
-  // Set the frame_idx_ map with indices of samples that are not empty
-  void SetFrameIdxs() {
-    frame_idx_.clear();
-    frame_idx_.reserve(in_shape_.num_samples());
-    for (int i = 0; i < in_shape_.num_samples(); ++i) {
-      if (volume(out_shape_.tensor_shape_span(i)) != 0 &&
-                   volume(in_shape_.tensor_shape_span(i)) != 0) {
-        frame_idx_.push_back(i);
+  // Assign each minibatch a range of frames in the original input/output TensorLists
+  void CalculateSourceSamples(const TensorListShape<> &original_shape, int first_spatial_dim) {
+    int64_t sample_id = 0;
+    int64_t frame_offset = 0;
+    for (auto &mb : minibatches_) {
+      auto v = original_shape[sample_id].num_elements();
+      while (v == 0) {
+        sample_id++;
+        v = original_shape[sample_id].num_elements();
+      }
+      mb.sample_offset = sample_id;
+      mb.frame_offset = frame_offset;
+      frame_offset = mb.frame_offset + mb.count;
+      int frames_n = num_frames(original_shape[sample_id], first_spatial_dim);
+      while (frame_offset >= frames_n) {
+        frame_offset -= frames_n;
+        if (++sample_id >= original_shape.num_samples()) {
+          break;
```

Comment:
Is it OK that the outerloop could spin further once we break out of the inner one?